src/providers.ts

```diff
@@ -79,28 +78,23 @@ const fetchBitbucketFocusViewData = async (token: ProviderToken) => {
 };
 
 const fetchBitbucketServerFocusViewData = async (token: ProviderToken) => {
-	// Bitbucket Server does not have the CORS header set to be able to make requests from the browser,
-	// so we proxy the request through the API.
-	const res = await fetch(`${GKDotDevUrl}/api/provider/bitbucket-server/proxy`, {
-		headers: {
-			Authorization: `Bearer ${token.accessToken}`,
-			XDestination: `${token.domain}/rest/api/latest/dashboard/pull-requests`,
-		},
+	const bitbucketServer = new BitbucketServer({
+		token: token.accessToken,
+		// Bitbucket Data Center does not have the CORS header set to be able to make requests from the browser,
+		// so we proxy the request through the API.
+		baseUrl: `${GKDotDevUrl}/api/provider/bitbucket-server/proxy/${encodeURIComponent(
+			`${token.domain?.replace(/\/+$/, '')}/rest/api/latest`,
+		)}`,
 	});
 
-	if (!res.ok) {
-		throw new Error('Failed to fetch Bitbucket Server pull requests');
-	}
+	const { data: providerUser } = await bitbucketServer.getCurrentUser();
 
-	const data = await res.json();
+	const { data: pullRequests } = await bitbucketServer.getPullRequestsForCurrentUser({});
 
 	return {
-		providerUser: data.user as Account,
-		pullRequests: (data.body.values as any[]).map(pullRequest => ({
-			...BitbucketServerUtils.restApiPullRequestToCommonPullRequest(pullRequest),
-			// Bitbucket Server PR ids are just the number, they are not unique across repos, so instead
-			// we use the PR url as the id.
-			id: pullRequest.links.self[0].href,
+		providerUser: providerUser,
+		pullRequests: pullRequests.map(pr => ({
```

Comment:
```suggestion
		pullRequests: (pullRequests || []).map(pr => ({
```
suggestion: maybe a guard would make sure it doesn't explode if the request fails